setup_thirdparty.sh

```diff
@@ -21,7 +21,7 @@ fi
 $TP_DIR/thirdparty/scripts/setup.sh $PYTHON_EXECUTABLE $LANGUAGE
 
 if [[ "$LANGUAGE" == "java" ]]; then
-    pushd thirdparty/build/arrow/java
+    pushd $TP_DIR/thirdparty/build/arrow/java
```

Comment:
While we are at it, can you also rename $TP_DIR to $ROOT_DIR? That's more appropriate, since it really is the root directory and not the ThirdParty_DIRectory.